### PR TITLE
Update readme

### PR DESCRIPTION
**Pull request by ellenmobify**
Status: **Ready for Review** 
Owner: Ellen 
Reviewers: @marlowpayne @RobotRogue @twangtina @gsaray @lilyzh @mobify-derrick 

## Changes
- Copies over all the nightwatch-commands documentation on http://adaptivejs.mobify.com/v2.0/docs/custom-nightwatchjs-api/ into this project's README
- Updates the README with documentation for new/changed commands
- Adds CONTRIBUTING.md 

## Jira Tickets:
- [x] https://mobify.atlassian.net/browse/CSOG-178

## Todos:
- [x] Proofread +1 

### Feedback:
- [x] add issue and PR templates
- [x] table whitespace

### CODE REVIEW

File: CONTRIBUTING.md [hunk 1]
```diff
@@ -0,0 +1,12 @@
+# Contributing to nightwatch-commands
+
+## Bugs, Feature Requests
+Submit an [issue](https://github.com/mobify/nightwatch-commands/issues).
+
+## Pull Requests
+
+1. Make a new branch from `develop`.
+1. Do your work.
+1. Update the README.
```
Comment (mobify-derrick) [inline CONTRIBUTING.md]: I would also suggest that we require test(s) for whatever is being added.

**Comment by mobify-derrick** [comment]
This is great so far. Given that GH released `issue` and `pull request` templates, I would like them to be added.

https://github.com/blog/2111-issue-and-pull-request-templates

Let's get some initial ones in and we can iterate on them.

File: CONTRIBUTING.md (hunk 1 above)
Comment (ellenmobify) [inline CONTRIBUTING.md]: Our tests don't currently work. Should we require tests now, or add the requirement after we fix the tests?

Comment (mobify-derrick) [inline CONTRIBUTING.md]: Good point. Lets move that to another PR once we get tests working again.

File: README.md [hunk 2]
```diff
@@ -15,3 +13,309 @@ To begin, clone this repository. `cd` into your chosen folder, and run the follo
...
 Linting may be run with the `grunt lint` command. Code is also linted automatically on [CircleCI](https://circleci.com/).
+
+## Assertions
+
+#### elementsCount(selector, expected, message, callback)
+
+The `elementsCount` assertion checks if the given selector is present the number of times that is expected for that selector to appear.
+
+Parameter Name | Parameter Type | Description
+--- | --- | ---
```
Comment (marlowpayne) [inline README.md]: This is a non-blocking thing, but with markdown tables like this whitespace and extra characters for the horizontal rules are ignored. This means you can space things out so that a maintainer can still get a good impression of the table while just viewing the source, like this (check this comment's source):

       Key        |        Value
----------------- | -------------------
       key1       |        value1
       key2       |        value2

Comment (mobify-derrick) [inline README.md]: @marlowpayne TIL thx, @ellenmobify if this is not a ton of work, I would actually like to implement this in this PR so that it is one less thing that we need to address later.

File: README.md [hunk 3]
```diff
@@ -15,3 +13,309 @@ To begin, clone this repository. `cd` into your chosen folder, and run the follo
...
+this.demoTest = function (client) {
+    browser.assert.templateName('home');
+};
+```
+
+## Commands
+
+#### get(url, callback)
+
+The `get` command combines the `url` and `waitUntilMobified` functions. This command uses the `browser.execute` function to run code within the client browser. It then waits for the `Mobify` object to be present on the page
```
Comment (mobify-derrick) [inline README.md]: It then waits for the `Mobify` or `Adaptive` ...

File: README.md [hunk 4]
```diff
@@ -15,3 +13,309 @@ To begin, clone this repository. `cd` into your chosen folder, and run the follo
...
+
+```
+this.demoTest = function (browser) {
+    browser.getMobifyEvaluatedData();
+};
+```
+
+#### htmlCapture(message, callback)
+
+The `htmlCapture` command saves desktop HTML to `tests/system/integration/fixtures`, for use in integration tests. This command should be run using an environment with a desktop user agent.
```
Comment (mobify-derrick) [inline README.md]: The `htmlCapture` is a helper command that saves a pages HTML to `tests/system/integration/fixtures`.
This command was designed to collect and save HTML files for use in integration tests. The initial design of his command was to be run using a nightwatch environment with a desktop user agent.

However this can still be to capture any HTML the user requires.

File: README.md [hunk 5]
```diff
@@ -15,3 +13,309 @@ To begin, clone this repository. `cd` into your chosen folder, and run the follo
...
+
+```
+this.demoTest = function (browser) {
+    browser.htmlCapture('cart.html');
+};
+```
+
+#### log(message, callback)
+
+The `log` command prints a message to the console. Use this command to add messages to test output.
```
Comment (mobify-derrick) [inline README.md]: Use this command to output a message in a test.

**Comment by mobify-derrick** [comment]
Once we merge this, I'll set up protected branches for the repo

File: .github/PULL_REQUEST_TEMPLATE.md [hunk 6]
```diff
@@ -0,0 +1,17 @@
...
+
+## Todos:
+- [] Passed linting check (run `grunt lint`)
+- [] Updated README
+
+### Feedback:
+_none so far_
+
+## How To Test
+- (note1)
```
Comment (mobify-derrick) [inline .github/PULL_REQUEST_TEMPLATE.md]: Let's add the lint command here as well

File: .github/PULL_REQUEST_TEMPLATE.md [hunk 7]
```diff
@@ -0,0 +1,17 @@
...
+Owner: 
+Reviewers: @ellenmobify @marlowpayne @mobify-derrick
+
+## Changes
+- (change1)
+- (change2)
+
+## Todos:
+- [] Passed linting check (run `grunt lint`)
+- [] Updated README
```
Comment (mobify-derrick) [inline .github/PULL_REQUEST_TEMPLATE.md]: CHANGELOG.md has been updated

File: .github/ISSUE_TEMPLATE.md [hunk 8]
```diff
@@ -0,0 +1,5 @@
+### Expected behaviour
```
Comment (mobify-derrick) [inline .github/ISSUE_TEMPLATE.md]: Can we add a ### Summary section at the top?

File: .github/PULL_REQUEST_TEMPLATE.md (hunk 7 above)
Comment (ellenmobify) [inline .github/PULL_REQUEST_TEMPLATE.md]: Added.

**Comment by mobify-derrick** [comment]
:+1: I think this ready to go. @marlowpayne any additional comments?

**Comment by marlowpayne** [comment]
This looks solid. :+1: nice work, @ellenmobify :shipit:
